src/ray/core_worker/task_manager.cc

```diff
@@ -433,6 +433,12 @@ void TaskManager::FailPendingTask(const TaskID &task_id,
     submissible_tasks_.erase(it);
     num_pending_tasks_--;
 
+    if (ray_error_info != nullptr) {
+      TaskFailureEntry entry(ray_error_info->exception());
```

Comment:
@rkooo567 I'm confused, why do we need to save this? We should already be storing the exception as the task's return value, so why do we need the additional cache for failure reasons?

IMO, it would be better to only have this in the raylet, which can flush worker errors to the GCS worker table. If the state API needs that info, it can join the worker table info with the task table info.